xml/resources-2.8.rng

```diff
@@ -0,0 +1,300 @@
+          <ref name="element-group"/>
+          <ref name="element-clone"/>
+          <ref name="element-master"/>
+          <ref name="element-bucket"/>
+        </choice>
+      </zeroOrMore>
+    </element>
+  </define>
+
+  <define name="element-primitive">
+    <element name="primitive">
+      <interleave>
+        <attribute name="id"><data type="ID"/></attribute>
+        <choice>
+          <group>
+            <ref name="element-resource-class"/>
+            <attribute name="type"><text/></attribute>
+          </group>
+          <attribute name="template"><data type="IDREF"/></attribute>
+        </choice>
+        <optional>
+          <attribute name="description"><text/></attribute>
+        </optional>
+        <ref name="element-resource-extra"/>
+        <ref name="element-operations"/>
+        <zeroOrMore>
+          <element name="utilization">
+            <externalRef href="nvset-1.3.rng"/>
+          </element>
+        </zeroOrMore>
+      </interleave>
+    </element>
+  </define>
+
+  <define name="element-template">
+    <element name="template">
+      <interleave>
+        <attribute name="id"><data type="ID"/></attribute>
+        <ref name="element-resource-class"/>
+        <attribute name="type"><text/></attribute>
+        <optional>
+          <attribute name="description"><text/></attribute>
+        </optional>
+        <ref name="element-resource-extra"/>
+        <ref name="element-operations"/>
+        <zeroOrMore>
+          <element name="utilization">
+            <externalRef href="nvset-1.3.rng"/>
+          </element>
+        </zeroOrMore>
+      </interleave>
+    </element>
+  </define>
+
+
+  <define name="element-bucket">
+    <element name="bucket">
+	<attribute name="id"><data type="ID"/></attribute>
+	<optional>
+          <attribute name="description"><text/></attribute>
+	</optional>
+        <choice>
+	  <group>
+	    <element name="docker">
+	      <attribute name="image"><text/></attribute>
+	      <optional>
+		<attribute name="replicas"><data type="integer"/></attribute>
+	      </optional>
+	      <optional>
+		<attribute name="replicas-per-host"><data type="integer"/></attribute>
+	      </optional>
+	      <optional>
+		<attribute name="masters"><data type="integer"/></attribute>
+	      </optional>
+	      <optional>
+		<attribute name="pcmk-remote-bin"> <text/></attribute>
+	      </optional>
+	      <optional>
+		<attribute name="options"><text/></attribute>
+	      </optional>
+	    </element>
+	  </group>
+	</choice>
+	<optional>
+	  <element name="network">
+	    <optional>
+	      <attribute name="ip-range-start"><text/></attribute>
+	    </optional>
+	    <optional>
+	      <attribute name="host-network"><text/></attribute>
+	    </optional>
+	    <optional>
+	      <attribute name="host-netmask"><data type="integer"/></attribute>
+	    </optional>
+	    <optional>
+	      <attribute name="docker-network"><text/></attribute>
+	    </optional>
```

Comment:
"docker-network" as a name seems out of place outside the docker tag. Would "container-network" work, or is this something really specific to docker?